tests/test_tuya.py

```diff
@@ -137,6 +138,73 @@ async def test_singleswitch_state_report(zigpy_device_from_quirk, quirk):
     assert switch_listener.attribute_updates[1][1] == OFF
 
 
+@pytest.mark.parametrize("quirk", (zhaquirks.tuya.ts0021.TS0021,))
+async def test_ts0021_switch(zigpy_device_from_quirk, quirk):
+    """Test tuya TS0021 2-gang switch."""
+
+    device = zigpy_device_from_quirk(quirk)
+
+    tuya_cluster = device.endpoints[1].tuya_manufacturer
+    switch_listener = ClusterListener(tuya_cluster)
+
+    ZCL_BTN1_SINGLE_PRESS = b"\tT\x06\x01$\x01\x04\x00\x01\x00"
+    hdr, args = tuya_cluster.deserialize(ZCL_BTN1_SINGLE_PRESS)
+    assert args.data.datapoints[0].dp == 1
+    assert args.data.datapoints[0].data.raw == b"\x00"
+    tuya_cluster.handle_message(hdr, args)
```

Comment:
You might be able to extract the data from the test itself and put it into a parametrize fixture. That might simplify the test a bit. It would just run multiple times for each of the messages and expected `raw` data and triggers.

If you need help with it, I might be able to do it soon-ish.

Reply:
I think I managed to parametrize most of it and I agree it looks much cleaner 👌